### PR TITLE
Buildbot password templating

### PR DESCRIPTION
**Pull request by stumped2**
Create private.py using a hash for passwords. Also do a bit of format cleanup.

### CODE REVIEW

File: modules/buildbot_asf/templates/private.py.erb [hunk 1]
```diff
@@ -1,40 +1,3 @@
...
-slave30Pwd = '<%= @slave30Pwd %>'
-slave31Pwd = '<%= @slave31Pwd %>'
-slave32Pwd = '<%= @slave32Pwd %>'
-slave33Pwd = '<%= @slave33Pwd %>'
-slave34Pwd = '<%= @slave34Pwd %>'
-slave36Pwd = '<%= @slave36Pwd %>'
-slave37Pwd = '<%= @slave37Pwd %>'
-bb-slave1Pwd = '<%= @bb-slave1Pwd %>'
+<% @master_list.keys.sort.each do |key| -%>
+<%= key %> = '<%= @master_list[key] %>'
```
Comment (dspt) [inline modules/buildbot_asf/templates/private.py.erb]: Does indentation here matter?

Comment (stumped2) [inline modules/buildbot_asf/templates/private.py.erb]: Nope. I also tested this locally and the end result was left aligned in the file and no puppet errors/warnings.

**Comment by stumped2** [comment]
@gmcdonald this look good to merge?

**Comment by gmcdonald** [comment]
The $master_list is empty.

I'm still a bit confused on what fetches the encrypted password from the eyaml.

The old way did this with a '<%= @slave1Pwd %>' etc etc but where is this now?

**Comment by stumped2** [comment]
@gmcdonald This is where I blame the github ui for being stupid. It's not listing out the changes in data/nodes/buildbot-vm.apache.org.eyaml, but it shows it as a changed file.

In data/nodes/buildbot-vm.apache.org.eyaml,  there is:
```
buildbot_asf::master_list:
  slave1Pwd: <gpg encrypted password>
  slave2Pwd: <gpgencrypted password>
  ...
```

So it's the hash.

**Pull request merged**